js_modules/dagit/packages/core/src/ticks/EvaluateSchedule.tsx

```diff
@@ -0,0 +1,400 @@
+          }
+        >
+          <div {...containerProps}>
+            <Button
+              style={{flex: 1}}
+              rightIcon={<Icon name="arrow_drop_down" />}
+              onClick={() => setIsTickSelectionOpen((isOpen) => !isOpen)}
+              data-testid={testId('tick-selection')}
+            >
+              {selectedTimestampRef.current.label}
+            </Button>
+          </div>
+        </Popover>
+      </SelectWrapper>
+    );
+  }, [
+    _selectedTimestamp,
+    containerProps,
+    data,
+    isTickSelectionOpen,
+    jobName,
+    name,
+    repoAddress,
+    shouldEvaluate,
+    userTimezone,
+    viewport.width,
+  ]);
+
+  const buttons = React.useMemo(() => {
+    if (!shouldEvaluate) {
+      return (
+        <>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button
+            data-testid={testId('evaluate')}
+            intent="primary"
+            onClick={() => {
+              setShouldEvaluate(true);
+            }}
+          >
+            Evaluate
+          </Button>
+        </>
+      );
+    } else {
+      return <Button onClick={onClose}>Close</Button>;
+    }
+  }, [onClose, shouldEvaluate]);
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      style={{width: '70vw', display: 'flex'}}
+      title={
+        <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+          <Icon name="schedule" />
+          <span>{name}</span>
+        </Box>
+      }
+    >
+      <DialogBody>
+        <div style={{minHeight: '300px'}}>{content}</div>
+      </DialogBody>
+      {buttons ? <DialogFooter topBorder>{buttons}</DialogFooter> : null}
+    </Dialog>
+  );
+};
+
+export const GET_SCHEDULE_QUERY = gql`
+  query getSchedule(
+    $scheduleSelector: ScheduleSelector!
+    $startTimestamp: Float
+    $ticksAfter: Int
+    $ticksBefore: Int
+  ) {
+    scheduleOrError(scheduleSelector: $scheduleSelector) {
+      __typename
+      ... on PythonError {
+        message
+        stack
+      }
+      ... on Schedule {
+        name
+        potentialTickTimestamps(
+          startTimestamp: $startTimestamp
+          upperLimit: $ticksAfter
+          lowerLimit: $ticksBefore
+        )
+      }
+    }
+  }
+`;
+
+const EvaluateScheduleContent: React.FC<{
+  repoAddress: RepoAddress;
+  name: string;
+  timestamp: number;
+  jobName: string;
+}> = ({repoAddress, name, timestamp, jobName}) => {
+  const [userTimezone] = React.useContext(TimezoneContext);
+  const [scheduleDryRunMutation] = useMutation<
+    ScheduleDryRunMutation,
+    ScheduleDryRunMutationVariables
+  >(
+    SCHEDULE_DRY_RUN_MUTATION,
+    React.useMemo(() => {
+      const repositorySelector = repoAddressToSelector(repoAddress);
+      return {
+        variables: {
+          selectorData: {
+            ...repositorySelector,
+            scheduleName: name,
+          },
+          timestamp,
+        },
+      };
+    }, [name, repoAddress, timestamp]),
```

Comment:
This is fine but it'd be kinda nice to pull this up to a top level `const scheduleDryRunVariables = React.useMemo(()=>` above the `useMutation` line to keep things standard looking.